pyproject.toml

```diff
@@ -3,3 +3,9 @@ supported_extensions = ["py"]
 skip = [".git/", "src/rdb/raft", "build", "install", "venv", "src/control/vendor/", "_topdir"]
 line_length = 99
 skip_gitignore = true
+
+[tool.codespell]
+skip = './src/control/vendor,./src/control/go.sum,./.git,./src/rdb/raft,./build,./install,./venv,./src/control/vendor/'
+quiet-level = 3
+ignore-words = 'ci/codespell.ignores'
+builtin = 'clear,rare,informal,names,en-GB_to_en-US'
```

Comment:
This lets us run codespell almost similar to the githook and the GHA. But unfortunately codespell doesn't seem to have any git option or way to skip files in .gitignore

Reply:
Shame.  I had lots of failures when I updated to get this githook, many of them were from 'docs/doxygen` which is not included in the source but is the default build location for the documentation.

Reply:
Do we need to add docs/doxygen to the skip here?

Reply:
I was able to just delete the directory and carry on.  Like you say best would be to support a .gitignore file.

An option might be to use `git ls-files` or similar which would mean we only needed to skip committed files which should be ignored, not non-committed files as well.

Reply:
I thought `git diff` already ignored non-committed files?

Reply:
Oh yeah, I'm not sure why it was even testing that dir then.  Let me check again.